nio/schemas.py

```diff
@@ -1114,6 +1115,37 @@ class Schemas(object):
         "additionalProperties": False,
     }
 
+    m_read = {
+        "type": "object",
+        "properties": {
+            "content": {
+                "type": "object",
+                "patternProperties": {
+                    EventIdRegex: {
+                        "type": "object",
+                        "properties": {
+                            "m.read": {
+                                "type": "object",
+                                "patternProperties": {
+                                    UserIdRegex: {
+                                        "type": "object",
+                                        "properties": {
+                                            "ts": {"type" : "integer"}
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+            "type": {"type": "string"},
+            "room_id": {"type": "string"}
+        },
+        "required": ["content", "type"],
+        "additionalProperties": False,
```

Comment:
I think we should allow additional properties for future compat, I know that many places disallow them but that should be fixed one of these days.

Reply:
Ah okay I'll change that